shell/browser/native_window_views.cc

```diff
@@ -157,6 +162,10 @@ NativeWindowViews::NativeWindowViews(const gin_helper::Dictionary& options,
                                      NativeWindow* parent)
     : NativeWindow(options, parent),
       root_view_(std::make_unique<RootView>(this)),
+      // It's hacky to use GetPlatformNameForTest to figure out if we run under
+      // X11 but that's the best worst way at the moment.
+      is_x11_(ui::OzonePlatform::GetInstance()->GetPlatformNameForTest() ==
```

Comment:
We generally avoid testing apis as they are prone to removal, I think this can be replaced with `#if BUILDFLAG(OZONE_PLATFORM_X11)`

Reply:
I don't think we can guard on build flag. Running under X11 is a runtime property, the same binary can run under wayland, X11, or other platforms supported by ozone layer.

Alternative approach here would be to use [`skia_can_fall_back_to_x11`](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/public/ozone_platform.h;bpv=1;bpt=1;l=133?q=GetPlatformNameForTest&ss=chromium&gsn=skia_can_fall_back_to_x11&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dsrc%2Fui%2Fozone%2Fpublic%2Fozone_platform.h%23cr2mt2XFdJF2t4pnh2ZuoaoFuKd13NmSrxYfYoW67zs) or add a patch for upstream that introduces new dedicated platform property like `electron_can_call_x11`

Reply:
Ah okay, makes sense. In that case why not use the non-test variant [ui::GetOzonePlatformName](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform_selection.h;l=17) ? Also I would still consider the build guards to use `BUILDFLAG(OZONE_PLATFORM_X11)` for X11 paths so that we allow configuring `ozone_platform_*` gn args.

Reply:
`ui/ozone/platform_selection.h` is a private header, that would need another patch to build files to make it public.

Using `BUILDFLAG(OZONE_PLATFORM_X11)` for X11 paths sounds good, I will send commit latter to do that.

Reply:
[//ui/ozone](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/BUILD.gn;l=244-251) component is built with public visibility, so it should be possible to include headers from there without any patch.

Reply:
Sure, the component is public, but [the header file is private](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/BUILD.gn;drc=62fc260067cf0c17d5416448675bfd73dbf6db5a;l=183), [those are the public headers](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/BUILD.gn;drc=62fc260067cf0c17d5416448675bfd73dbf6db5a;l=192-194).

To make the point: when I add this include, `e d gn check out/Testing //electron:electron_lib` returns:
```
ERROR at //electron/shell/browser/native_window_views.cc:67:11: Including a private header.
#include "ui/ozone/platform_selection.h"
          ^----------------------------
This file is private to the target //ui/ozone:platform
```

Reply:
If we don't want to use test functions, and I will have to patch something, I've wend down the path of IMHO the cleanest solution and introduced `electron_can_call_x11` property.

Reply:
We actually include private headers when needed in the `//electron_lib` target, you can grep for `// nogncheck` in the code. If a target is visible then it should not be an issue to use the private headers, as long as it is ensured that the target is built. Now that I think about it, we could have also added the `//  nogncheck` for `gtk_util.h` header instead of the patch, will look into that in a follow-up.

> I've wend down the path of IMHO the cleanest solution and introduced electron_can_call_x11 property.

I like this 👍